src/screencast/wlr_screencast.c

```diff
@@ -580,7 +584,19 @@ static void wlr_registry_handle_remove(void *data, struct wl_registry *reg,
 	struct xdpw_screencast_context *ctx = data;
 	struct xdpw_wlr_output *output = xdpw_wlr_output_find(ctx, NULL, id);
 	if (output) {
-		wlr_remove_output(output);
+		logprint(DEBUG, "wlroots: output removed (%s)", output->name);
+		output->removed = true;
+		bool used = false;
+		struct xdpw_screencast_instance *cast, *tmp;
+		wl_list_for_each_safe(cast, tmp, &ctx->screencast_instances, link) {
+			if (cast->target_output == output) {
+				xdpw_screencast_instance_teardown(cast);
```

Comment:
I think that if you just set `cast->target_output = NULL` if it matches the removed output, and then check inside the screencast instance if it has an output before creating a frame, you should be fine. But for good measure, you could also call `zwlr_screencopy_v1_frame_destroy` here on any frames that have already been created.

Reply:
> you could also call `zwlr_screencopy_v1_frame_destroy` here on any frames that have already been created.

That was to easy.... thanks a lot!